### PR TITLE
Add acceptance tests for running chef-client with the FIPS mode

### PR DESCRIPTION
**Pull request by jaym**
Adding a test to the accepting framework for FIPS. When a package gets built, it will toggle FIPS mode for the platforms we care about and run the rspec tests. This keeps us from mucking around with the test matrix in jenkins.

### CODE REVIEW

**Comment by jaym** [comment]
cc @ksubrama @chefsalim @jkeiser @sersut @btm

**Comment by ksubrama** [comment]
:+1:

**Comment by btm** [comment]
:+1:

File: acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml [hunk 1]
```diff
@@ -38,8 +38,10 @@ fedora-21
 # freebsd-8
 # ubuntu-12.04
 # centos-7
-# centos-6
-
+  - name: centos-6
+    driver:
+      box: opscode-centos-6.6
+      box_url: http://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_centos-6.6_chef-provisionerless.box
```
Comment (schisamo) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: Why are we pulling in a special (OLD) box here? `centos-6.6` should pull down the correct bento box.

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: I want to say it was because it matches the one in ec2

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: i lied, ec2 uses 6.5

Comment (schisamo) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: Take a look at: https://github.com/chef-cookbooks/omnibus/blob/master/.kitchen.yml#L11-L35

You should not have to specify a box name or URL.

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: I want the ec2 name and vagrant name to be the same, which is `centos-6`

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]:
```
- name: centos-6
    driver:
      box: centos-6.7
```
does not work

Comment (schisamo) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: Try:

```yaml
- name: centos-6
   driver:
      box: bento/centos-6.7
```

See this kitchen-vagrant change for more:
https://github.com/test-kitchen/kitchen-vagrant/commit/5697d9fd77617b41daf4b0f5165547449b856a43

If we would publish the CentOS Bento boxes to Atlas as `centos-MAJOR` you could drop even specifying a specific box name.

/cc @cheeseplus

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: nope, seems to ignore the box. Maybe that's what it's using to name it locally?
```
       ==> default: Adding box 'bento/centos-6.7' (v0) for provider: virtualbox
           default: Downloading: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_centos-6_chef-provisionerless.box
```

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: For some reason, it's filling in the url in the Vagrantfile. It looks like:

```
Vagrant.configure("2") do |c|
  c.berkshelf.enabled = false if Vagrant.has_plugin?("vagrant-berkshelf")
  c.vm.box = "bento/centos-6.7"
  c.vm.box_url = "https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_centos-6_chef-provisionerless.box"
  c.vm.hostname = "fips-centos-6"
  c.vm.synced_folder ".", "/vagrant", disabled: true
  c.vm.provider :virtualbox do |p|
    p.customize ["modifyvm", :id, "--cpus", "2"]
    p.customize ["modifyvm", :id, "--memory", "1024"]
  end
end
```

Comment (jaym) [inline acceptance/.shared/kitchen_acceptance/.kitchen.vagrant.yml]: https://github.com/test-kitchen/kitchen-vagrant/blob/v0.19.0/lib/kitchen/driver/vagrant.rb#L104-L113

kitchen-vagrant needs a new release. Going to point the Gemfile at github

**Comment by schisamo** [comment]
:+1:

**Pull request merged**